airbyte-scheduler/persistence/src/main/java/io/airbyte/scheduler/persistence/job_factory/OAuthConfigSupplier.java

```diff
@@ -0,0 +1,78 @@
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package io.airbyte.scheduler.persistence.job_factory;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import io.airbyte.commons.json.Jsons;
+import io.airbyte.config.DestinationOAuthParameter;
+import io.airbyte.config.SourceOAuthParameter;
+import io.airbyte.config.persistence.ConfigRepository;
+import io.airbyte.validation.json.JsonValidationException;
+import java.io.IOException;
+import java.util.Comparator;
+import java.util.UUID;
+
+// TODO chris: move to oauth module
+public class OAuthConfigSupplier {
+
+  final private ConfigRepository configRepository;
+
+  public OAuthConfigSupplier(ConfigRepository configRepository) {
+    this.configRepository = configRepository;
+  }
+
+  public JsonNode injectSourceOAuthParameters(UUID sourceDefinitionId, UUID workspaceId, JsonNode sourceConnectorConfig)
+      throws JsonValidationException, IOException {
+    configRepository.listSourceOAuthParam().stream()
+        .filter(p -> sourceDefinitionId.equals(p.getSourceDefinitionId()))
+        .filter(p -> p.getWorkspaceId() == null || workspaceId.equals(p.getWorkspaceId()))
+        // we prefer params specific to a workspace before global ones (ie workspace is null)
+        .min(Comparator.comparing(SourceOAuthParameter::getWorkspaceId, Comparator.nullsLast(Comparator.naturalOrder()))
+            .thenComparing(SourceOAuthParameter::getOauthParameterId))
+        .ifPresent(sourceOAuthParameter -> injectJsonNode((ObjectNode) sourceConnectorConfig, (ObjectNode) sourceOAuthParameter.getConfiguration()));
+    return sourceConnectorConfig;
+  }
+
+  public JsonNode injectDestinationOAuthParameters(UUID destinationDefinitionId, UUID workspaceId, JsonNode destinationConnectorConfig)
+      throws JsonValidationException, IOException {
+    configRepository.listDestinationOAuthParam().stream()
+        .filter(p -> destinationDefinitionId.equals(p.getDestinationDefinitionId()))
+        .filter(p -> p.getWorkspaceId() == null || workspaceId.equals(p.getWorkspaceId()))
+        // we prefer params specific to a workspace before global ones (ie workspace is null)
+        .min(Comparator.comparing(DestinationOAuthParameter::getWorkspaceId, Comparator.nullsLast(Comparator.naturalOrder()))
+            .thenComparing(DestinationOAuthParameter::getOauthParameterId))
+        .ifPresent(destinationOAuthParameter -> injectJsonNode((ObjectNode) destinationConnectorConfig,
+            (ObjectNode) destinationOAuthParameter.getConfiguration()));
+    return destinationConnectorConfig;
+  }
+
+  private static void injectJsonNode(ObjectNode config, ObjectNode fromConfig) {
+    for (String key : Jsons.keys(fromConfig)) {
+      config.set(key, fromConfig.get(key));
```

Comment:
If the OAuth parameter was already part of the connector config, should we still inject and overwrite it with the one from the instance-wide values? 

or should they be a special behavior here? (adding only if does not exists yet)